libs/designer-ui/src/lib/editor/base/utils/__test__/helper.spec.ts

```diff
@@ -33,3 +33,49 @@ describe('Helper functions', () => {
     ]);
   });
 });
+
+describe('getDropdownOptionsFromOptions', () => {
+  it('returns options directly when they are available', () => {
+    const editorOptions = {
+      options: [
+        { key: '1', value: 'Option 1', displayName: 'Option 1' },
+        { key: '2', value: 'Option 2', displayName: 'Option 2' },
+      ],
+    };
+    const result = getDropdownOptionsFromOptions(editorOptions);
+    expect(result).toEqual(editorOptions.options);
+  });
+
+  it('returns options when they are nested in an object', () => {
+    const editorOptions = {
+      options: {
+        nested: [
+          { key: '1', value: 'Option 1', displayName: 'Option 1' },
+          { key: '2', value: 'Option 2', displayName: 'Option 2' },
+        ],
+      },
+    };
+    const result = getDropdownOptionsFromOptions(editorOptions);
+    expect(result).toEqual(editorOptions.options.nested);
+  });
+
+  it('handles cases where displayName is not a string', () => {
+    const editorOptions = {
+      options: [
+        { key: '1', value: 'Option 1', displayName: { text: 'Option 1' } },
+        { key: '2', value: 'Option 2', displayName: ['Option 2'] },
+      ],
```

Comment:
something is not right, value can be any type for displayName is always a string

Reply:
yeah this was a check because there was an issue where some connectors were giving us display names where they were non-string types, which was crashing our designer